sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -178,6 +178,24 @@ case class DataSourceResolution(
       val aliased = delete.tableAlias.map(SubqueryAlias(_, relation)).getOrElse(relation)
       DeleteFromTable(aliased, delete.condition)
 
+    case ShowNamespacesStatement(None, pattern) =>
+      defaultCatalog match {
```

Comment:
I think this should be `currentCatalog` instead. @cloud-fan, do you agree?

Reply:
yea

Reply:
Let's implement switching the current catalog first, otherwise we are not able to test it.

Reply:
@imback82 are you working on it?

Reply:
yes, I am working on `USE NAMESPACE`

Reply:
I should be able to send out the PR sometime tomorrow.